jplag.endToEndTesting/src/test/java/de/jplag/end_to_end_testing/JavaEndToEndTest.java

```diff
@@ -100,15 +143,28 @@ void tokenGenerationLevelTest(String[] testClassNames, int testIdentifier) throw
      * @param testIdentifier name of the testId to load and identify the stored results
      * @throws IOException is thrown in case of problems with copying the plagiarism classes
      * @throws ExitException in case the plagiarism detection with JPlag is preemptively terminated would be of the test.
+     * @throws NoSuchAlgorithmException when no hash algorithm could be found
+     * @throws NameNotFoundException if the no filenames cloud be found in the JPlagCOmparison object
      */
-    private void runJPlagTestSuite(String[] testClassNames, int testIdentifier) throws IOException, ExitException {
-        String functionName = StackWalker.getInstance().walk(stream -> stream.skip(1).findFirst().get()).getMethodName();
+    private void runJPlagTestSuite(String[] testClassNames, String functionName)
+            throws IOException, ExitException, NoSuchAlgorithmException, NameNotFoundException {
+        // String functionName = StackWalker.getInstance().walk(stream -> stream.skip(1).findFirst().get())
+        // .getMethodName();
         TestCaseModel testCaseModel = jplagTestSuiteHelper.createNewTestCase(testClassNames, functionName);
         JPlagResult jplagResult = new JPlag(testCaseModel.getJPlagOptionsFromCurrentModel()).run();
+        List<JPlagComparison> currentJPlagComparison = jplagResult.getAllComparisons();
+        jplagTestSuiteHelper.saveTemporaryResult(currentJPlagComparison, functionName);
 
-        for (JPlagComparison jPlagComparison : jplagResult.getAllComparisons()) {
-            assertEquals(testCaseModel.getCurrentJsonModel().getResultModelById(testIdentifier).getResultSimilarity(), jPlagComparison.similarity(),
-                    "The JPlag results [similarity] do not match the stored values!");
+        for (JPlagComparison jPlagComparison : currentJPlagComparison) {
+            String hashCode = jplagTestSuiteHelper.getTestIdentifier(jPlagComparison);
+            ResultModel resultModel = testCaseModel.getCurrentJsonModel().getResultModelById(hashCode);
+            assertNotNull(resultModel, "No stored result could be found for the identifier! " + hashCode);
+            assertEquals(resultModel.getMinimalSimilarity(), jPlagComparison.minimalSimilarity(),
+                    "The JPlag results [minimalSimilarity] do not match the stored values!");
+            assertEquals(resultModel.getMaximalSimilarity(), jPlagComparison.maximalSimilarity(),
+                    "The JPlag results [maximalSimilarity] do not match the stored values!");
+            assertEquals(resultModel.getNumberOfMatchedTokens(), jPlagComparison.getNumberOfMatchedTokens(),
+                    "The JPlag results [numberOfMatchedTokens] do not match the stored values!");
```

Comment:
It would be great to first collect all incorrect results. For example a `List<String> validationErrors` could be filled with error descriptions for incorrect values. Then at the end that array is checked for not empty. This has the benefit that the test failure description logs all incorrect values instead of only the first incorrect one.
I.e. test failure description:
```
There were <count> validation error(s):
- min similarity was <a> but expected <b>
- max similarity was <c> but expected <d>
```

Reply:
done
looks like this:
```
org.opentest4j.AssertionFailedError: 
There were <3> validation error(s):
minimalSimilarity was 99.0 but expected 100.0
maximalSimilarity was 99.0 but expected 100.0
numberOfMatchedTokens was 54 but expected 56
 ==> expected: <true> but was: <false>
```